src/briefcase/integrations/base.py

```diff
@@ -8,15 +8,29 @@
 from collections import defaultdict
 from collections.abc import Mapping
 from pathlib import Path
-from typing import DefaultDict
+from typing import TYPE_CHECKING, DefaultDict
 
 import requests
 from cookiecutter.main import cookiecutter
 
 from briefcase.config import AppConfig
 from briefcase.console import Console, Log
-from briefcase.integrations.download import Download
-from briefcase.integrations.subprocess import Subprocess
+
+if TYPE_CHECKING:
+    # Tools are imported only for type checking
+    # to avoid circular import errors.
+    import git as git_
```

Comment:
I was almost ready to approve this PR... and then I [remembered something](https://github.com/beeware/briefcase/pull/964/files#diff-d705b8400e278985802862242e3ab69651e42f2317d9ef7f4351428ccac4a751R8-R17).

That opens up a bunch of questions:
1. On what condition will `TYPE_CHECKING` be true?
2. Is it possible for the user to have `TYPE_CHECKING` be true and *not* have a working git install?
3. Is there any meaningful fallback/error recovery to use in this case?

Reply:
> I was almost ready to approve this PR... and then I [remembered something](https://github.com/beeware/briefcase/pull/964/files#diff-d705b8400e278985802862242e3ab69651e42f2317d9ef7f4351428ccac4a751R8-R17).
> 
> That opens up a bunch of questions:
> 
> 1. On what condition will `TYPE_CHECKING` be true?

`TYPE_CHECKING` probably shouldn't ever be `True`; even for type checkers like mypy, they should only [assume](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) `TYPE_CHECKING` is the same as `True`.

> 2. Is it possible for the user to have `TYPE_CHECKING` be true and _not_ have a working git install?

It's probably possible if something monkeypatched `typing.TYPE_CHECKING` at interpreter startup. But that definitely isn't the intended way to "assume" `TYPE_CHECKING` is `True` when relevant. I really hope the stronger promise of the design of `TYPE_CHECKING` is that it is always `False` at runtime.

> 3. Is there any meaningful fallback/error recovery to use in this case?

Not currently. But if we wanted to ensure this was avoided, we could change `if TYPE_CHECKING:` to simply `if False:`. Although, for instance, `pytest` explicitly [moved](https://github.com/pytest-dev/pytest/pull/6435) from `if False:` once Python 3.5 could use `typing.TYPE_CHECKING`.